rust/macros/module.rs

```diff
@@ -512,17 +514,28 @@ pub(crate) fn module(ts: TokenStream) -> TokenStream {
             static THIS_MODULE: kernel::ThisModule = unsafe {{ kernel::ThisModule::from_ptr(core::ptr::null_mut()) }};
 
             // Loadable modules need to export the `{{init,cleanup}}_module` identifiers
+            /// Tries to initialize the module and then unsynchronously sets `__MOD` to the initialized module.
+            /// Internally calls `__init()`.
+            ///
+            /// # Safety
+            ///
+            /// Should not be called unsynchronously with `cleanup_module()`, which unsets `__MOD`.
             #[cfg(MODULE)]
             #[doc(hidden)]
             #[no_mangle]
-            pub extern \"C\" fn init_module() -> kernel::c_types::c_int {{
+            pub unsafe extern \"C\" fn init_module() -> kernel::c_types::c_int {{
                 __init()
             }}
 
+            /// Exits the module by unsynchronously invoking `drop()` on `__MOD`. Internally calls `__exit()`.
+            ///
+            /// # Safety
+            ///
+            /// Should not be called unsynchronously with `init_module()`, which initializes `__MOD`.
```

Comment:
After calling this function it isn't valid to call into the kernel module anymore I believe.